internal/controller/helmrelease_controller.go

```diff
@@ -832,13 +867,20 @@ func (r *HelmReleaseReconciler) waitForHistoryCacheSync(obj *v2.HelmRelease) wai
 }
 
 func isSourceReady(obj sourcev1.Source) (bool, string) {
+	if o, ok := obj.(*sourcev1.ExternalArtifact); ok {
+		if obj.GetArtifact() == nil {
```

Comment:
Is this not also checked for the other SC APIs somewhere? It's kinda weird to be type-asserting only one of the APIs here

Reply:
Two reasons:
- ExternalArtifact does not implement the `fluxcd/ptg/runtime.Getter` interface 
- Reconcilers shouldn't care if the source is ready or not, they should only care if an artifact is available

Reply:
I made ExternalArtifact implement `Getter` but the 2nd point still stands, unlike the native sources that HC may generate, the external artifact can have a valid chart even when not ready, KC does this for all sources.